lib/matplotlib/colors.py

```diff
@@ -62,55 +65,201 @@
 import numpy as np
 from numpy import ma
 import matplotlib.cbook as cbook
-from ._color_data import XKCD_COLORS, CSS4_COLORS
+from ._color_data import BASE_COLORS, CSS4_COLORS, XKCD_COLORS
 
-# for back copatibility
-cnames = CSS4_COLORS
 
-COLOR_NAMES = {'xkcd': XKCD_COLORS,
-               'css4': CSS4_COLORS}
+class _ColorMapping(dict):
+    def __init__(self, mapping):
+        super(_ColorMapping, self).__init__(mapping)
+        self._cache = {}
```

Comment:
seaborn probably should _not_ do that, but that is a different discussion.  At least it is limited to the single letter colors.

That use-case should be eliminated by a combination of setting the color cycle + the `'CN'` color specification.